File: js/counter.js
// See also https://www.codewars.com/kata/reviews/57cb12de35ef62f92d000141/groups/57cb21fd58da9ee0d7000071
//dict object
function counter(e){
  const C = {}
  for (const i of e) C[i]=(C[i]||0)+1
  return C
}
const counter = require('lodash').countBy;
//on the fly
counter(n,arr){return arr.filter(e=>e==n).length}

//https://www.codewars.com/kata/reviews/59106fc3c37bca02fc000981/groups/5f80d9ed86498900011ee79d
const cnt=a=>a.reduce((o,n)=>(o[n]=o[n]+1||1,o), {});

File: cpp/counter.cpp
//see https://www.codewars.com/kata/5270d0d18625160ada0000e4/solutions/cpp  
unsigned c[size] = {};
  for (auto d : iterable)
    c[d]++;

//https://www.codewars.com/kata/554ca54ffa7d91b236000023/solutions/cpp

#include <unordered_map>
  std::unordered_map<int, unsigned> count;
  for (int k: arr) count[k]++;
